triton-vm/benches/initialize_array.rs

```diff
@@ -0,0 +1,90 @@
+use std::mem::MaybeUninit;
+
+use criterion::criterion_group;
+use criterion::criterion_main;
+use criterion::Criterion;
+use ndarray::prelude::*;
+use num_traits::One;
+use num_traits::Zero;
+use rand::rngs::StdRng;
+use rand::thread_rng;
+use rand::Rng;
+use rand::RngCore;
+use rand::SeedableRng;
+use twenty_first::prelude::*;
+
+criterion_main!(benches);
+criterion_group!(
+    name = benches;
+    config = Criterion::default().sample_size(10);
+    targets = init_array<{1 << 10}>,
+              init_array<{1 << 12}>,
+              init_array<{1 << 14}>,
+              init_array<{1 << 16}>,
+              init_array<{1 << 18}>,
+              init_array<{1 << 20}>,
+              init_array<{1 << 22}>,
+);
+
+fn init_array<const N: usize>(c: &mut Criterion) {
+    const W: usize = 50;
+    let mut group = c.benchmark_group(format!("initialize_array_{N}x{W}"));
+    let mut rng: StdRng = SeedableRng::from_seed(thread_rng().gen());
+    let mut matrix = Array2::<XFieldElement>::zeros((1, 1).f());
+    group.bench_function("zeros", |b| {
+        b.iter(|| {
+            matrix = Array2::<XFieldElement>::zeros((N, W).f());
+            set_ones::<N, W>(&mut rng, &mut matrix);
+        })
+    });
+    group.bench_function("into_shape", |b| {
+        b.iter(|| {
+            matrix = into_shape::<N, W>();
+            set_ones::<N, W>(&mut rng, &mut matrix);
+        })
+    });
+    group.bench_function("set_len", |b| {
+        b.iter(|| {
+            matrix = set_len::<N, W>();
+            set_ones::<N, W>(&mut rng, &mut matrix);
+        })
+    });
+    group.bench_function("maybe_uninit", |b| {
+        b.iter(|| {
+            matrix = maybe_uninit::<N, W>();
+            set_ones::<N, W>(&mut rng, &mut matrix);
+        })
+    });
+    group.finish();
+}
+
+fn set_ones<const H: usize, const W: usize>(rng: &mut StdRng, matrix: &mut Array2<XFieldElement>) {
+    assert!(!matrix.is_standard_layout());
+    for _ in 0..1000 {
+        let r = (rng.next_u32() as usize) % H;
+        let c = (rng.next_u32() as usize) % W;
```

Comment:
The `rand` crate provides methods specifically for this case.

```suggestion
        let r = rng.gen_range(0..H);
        let c = rng.gen_range(0..W);
```

This means that neither `as` casting nor modulo operations are necessary (on your end, anyway).